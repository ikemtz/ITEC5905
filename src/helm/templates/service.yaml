apiVersion: v1
kind: Service
metadata:
  name: {{ include "itec5905.fullname" . }}
  labels:
    {{- include "itec5905.labels" . | nindent 4 }}
spec:
  type: {{ .Values.service.type }}
  ports:
    - port: {{ .Values.service.port }}
      targetPort: http
      protocol: TCP
      name: http
  selector:
    {{- include "itec5905.selectorLabels" . | nindent 4 }}
