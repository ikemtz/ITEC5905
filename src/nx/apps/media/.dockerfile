FROM bitnami/node
EXPOSE 3000
WORKDIR /app
RUN npm i tslib @nestjs/platform-express @nestjs/common @nestjs/core @nestjs/swagger @pinata/sdk 
COPY ["./dist/apps/media/", "/app/"]
CMD node main.js