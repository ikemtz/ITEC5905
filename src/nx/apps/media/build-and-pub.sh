#!/bin/bash
set -ex
npx nx serve media & sleep 30
ps | grep node |  awk '{print $1}' | xargs kill
cp ./swagger.json ./src/assets
npx nx build media --configuration=development

docker build --pull \
    --rm \
    -f "./.dockerfile" \
    -t ikemtz/itec5905-media:latest ../../

docker image push ikemtz/itec5905-media:latest 

kubectl rollout restart deployment/media-webapi