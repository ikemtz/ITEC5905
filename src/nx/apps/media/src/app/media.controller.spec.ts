import { Test, TestingModule } from '@nestjs/testing';

import { MediaController } from './media.controller';
import { PinataService } from './pinata.service';
import { faker } from '@faker-js/faker';

describe('MediaController', () => {
  let app: TestingModule;

  beforeAll(async () => {
    app = await Test.createTestingModule({
      controllers: [MediaController],
      providers: [{
        provide: PinataService,
        useValue: {
          getFiles: jest.fn((x) => Promise.resolve([{
            id: x,
            ipfs_pin_hash: 'ðŸ˜ŽðŸ˜Ž',
            size: 1001,
            date_pinned: '1/1/2023',
            metadata: {
              refId: `1234-dabe`,
              refName: `Modern Bronze Car`,
              name: `asdf.json`,
            }
          }])),
          uploadFile: jest.fn((x) => Promise.resolve({ id: x, IpfsHash: 'ðŸ˜ŽðŸ˜Ž', Timestamp: 234239 })),
        }
      }],
    }).compile();
  });

  it('should Get Files', async () => {
    const appController = app.get<MediaController>(MediaController);
    const pinataService = app.get<PinataService>(PinataService);
    const result = await appController.getFiles('');
    expect(result[0]).toMatchSnapshot({
      createdOnUtc: expect.any(Date)
    });
    expect(pinataService.getFiles).toHaveBeenNthCalledWith(1, '');
  });

  it('should Upload Files', async () => {
    const appController = app.get<MediaController>(MediaController);
    const pinataService = app.get<PinataService>(PinataService);
    const result = await appController.uploadFile({
      fileName: 'goodTimes.exe',
      content: 'ðŸ˜’',
      referenceId: faker.commerce.productName(),
      referenceName: faker.commerce.productDescription(),
    });
    expect(result).toMatchSnapshot({
      createdOnUtc: expect.any(Date)
    });
    expect(pinataService.uploadFile).toBeCalledTimes(1);
  });
});
