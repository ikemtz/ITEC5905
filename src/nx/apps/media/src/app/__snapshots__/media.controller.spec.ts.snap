// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`MediaController should Get Files 1`] = `
{
  "createdOnUtc": Any<Date>,
  "fileName": "asdf.json",
  "fileSize": 1001,
  "ipfsHash": "ðŸ˜ŽðŸ˜Ž",
  "referenceId": "1234-dabe",
  "referenceName": "Modern Bronze Car",
}
`;

exports[`MediaController should Upload Files 1`] = `
{
  "createdOnUtc": Any<Date>,
  "fileSize": undefined,
  "ipfsHash": "ðŸ˜ŽðŸ˜Ž",
}
`;
